pkg/syncer/apiimporter.go

```diff
@@ -189,22 +216,22 @@ func (i *APIImporter) ImportAPIs(ctx context.Context) {
 			clusterctl.GetGVRForLocationInLogicalClusterIndexKey(i.location, i.logicalClusterName, gvr),
 		)
 		if err != nil {
-			klog.Errorf("error pulling CRDs: %v", err)
+			logger.Error(err, "error pulling CRDs")
 			continue
 		}
 		if len(objs) > 1 {
-			klog.Errorf("There should be only one APIResourceImport of GVR %s for location %s in logical cluster %s, but there was %d", gvr.String(), i.location, i.logicalClusterName, len(objs))
+			logger.Error(fmt.Errorf("failed to get APIResourceImport"), "There should be only one APIResourceImport", "gvr", gvr.String(), "location", i.location, "cluster", i.logicalClusterName, "obejcts", len(objs))
```

Comment:
Here also - when you add the location to the logger at the top  level, you never need to add it again